cpp/thirdparty/versions.txt

```diff
@@ -32,14 +32,17 @@ ARROW_BOOST_BUILD_VERSION=1.71.0
 ARROW_BROTLI_BUILD_VERSION=v1.0.7
 ARROW_BZIP2_BUILD_VERSION=1.0.8
 ARROW_CARES_BUILD_VERSION=1.16.1
+ARROW_CRC32C_BUILD_VERSION=1.0.6
 ARROW_GBENCHMARK_BUILD_VERSION=v1.5.2
+ARROW_GCPSDK_BUILD_VERSION=v1.20.0
 ARROW_GFLAGS_BUILD_VERSION=v2.2.2
 ARROW_GLOG_BUILD_VERSION=v0.4.0
 ARROW_GRPC_BUILD_VERSION=v1.29.1
 ARROW_GTEST_BUILD_VERSION=1.8.1
 ARROW_JEMALLOC_BUILD_VERSION=5.2.1
 ARROW_LZ4_BUILD_VERSION=v1.9.2
 ARROW_MIMALLOC_BUILD_VERSION=v1.6.4
+ARROW_NLOHMANN_JSON_BUILD_VERSION=v3.9.0
```

Comment:
Should URLs for these new dependencies also be added to `DEPENDENCIES` below? cf. ARROW-6407 ARROW-6333

Reply:
Whoops, thanks! I missed that. I've added the URLs for the dependencies I've inserted into `versions.txt`.

However, I've left out the backup URLs that use the ursa-labs archive in `ThirdPartyDependencies.cmake`, which I understand will be derived from the `versions.txt` URL eventually (if this gets merged). Let me know if its appropriate to link add these URLs.

Reply:
No, it's fine, don't add URLs that don't exist